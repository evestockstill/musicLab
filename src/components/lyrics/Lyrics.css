@import '../App.css';

.SongTitle {
  font-size: 4rem;
  padding-top: 3rem;
  font-family: 'Permanent Marker', cursive;
  text-align: center;
  color: $black;
}
.lyricsBox {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 3rem;

  .lyrics {
    display: flex;
    justify-content: center;
    align-items: center;
    font-family: 'Caveat', cursive;
    margin: 1rem auto;
    font-size: 2.5rem;
    letter-spacing: 2px;
    height: auto;
    width: 100rem;
    background: $green2;
    margin: 2rem;
    padding: 2rem;
    border-width: 1rem;
    border-style: solid;
    border-image: linear-gradient(to bottom, $orange, rgba(0, 0, 0, 0)) 1 100%;
  }
}
.spinner {
  width: 40px;
  height: 40px;
  position: relative;
  margin: 100px auto;
}
.bounce1,
.bounce2 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: #333;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-animation: sk-bounce 2s infinite ease-in-out;
  animation: sk-bounce 2s infinite ease-in-out;
}
.bounce2 {
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}
@-webkit-keyframes sk-bounce {
  0%,
  100% {
    -webkit-transform: scale(0);
  }
  50% {
    -webkit-transform: scale(1);
  }
}
@keyframes sk-bounce {
  0%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  50% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
.error {
  display: flex;
  justify-content: center;
  font-size: 4rem;
  padding: 3rem;
  margin-top: 4rem;
  font-family: 'Permanent Marker', cursive;

}
