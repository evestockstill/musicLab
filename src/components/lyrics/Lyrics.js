import React from 'react';
import PropTypes from 'prop-types';
import styles from './Lyrics.css';
import { useLyrics } from '../../hooks/lyrics';

const Lyrics = ({ match }) => {
  const { lyrics } = useLyrics(match.params.artistName, match.params.songId);
  return (
    <>
      <h2 className={styles.SongTitle}>Lyrics for {match.params.songId}</h2>
      <div className={styles.lyricsBox}>
        <p className={styles.lyrics}>{lyrics}</p>
      </div>
    </>
  );
};

Lyrics.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      artistName: PropTypes.string.isRequired,
      songId: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Lyrics;
