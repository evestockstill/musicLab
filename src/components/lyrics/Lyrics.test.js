import React from 'react';
import { shallow } from 'enzyme';
import Lyrics from './Lyrics';

describe('Lyrics component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Lyrics match={{ params: { id: '1' } }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
