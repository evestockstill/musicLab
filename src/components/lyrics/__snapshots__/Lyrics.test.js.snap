// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics component matches a snapshot 1`] = `
<div
  className="spinner"
>
  <div
    className="bounce1"
  />
  <div
    className="bounce2"
  />
</div>
`;
